rcl/test/rcl/test_init.cpp

```diff
@@ -98,65 +100,147 @@ struct FakeTestArgv
+  rcl_init_options_t init_options = rcl_get_zero_initialized_init_options();
+  rcl_ret_t ret = rcl_init_options_init(&init_options, rcl_get_default_allocator());
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+    EXPECT_EQ(RCL_RET_OK, rcl_init_options_fini(&init_options)) << rcl_get_error_string().str;
+  });
+
+  {
+    // If argc is not 0, but argv is, it should be an invalid argument.
+    rcl_context_t context = rcl_get_zero_initialized_context();
+    ret = rcl_init(42, nullptr, &init_options, &context);
+    EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+    rcl_reset_error();
+    ASSERT_FALSE(rcl_context_is_valid(&context));
+  }
   {
+    // If argc is not 0, argv is not null but contains one, it should be an invalid argument.
+    rcl_context_t context = rcl_get_zero_initialized_context();
+    const char * null_args[] = {"some-arg", nullptr};
+    ret = rcl_init(2, null_args, &init_options, &context);
+    EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+    rcl_reset_error();
+    ASSERT_FALSE(rcl_context_is_valid(&context));
+  }
+  {
+    // If argc is less than 1, argv is not null, it should be an invalid argument.
+    rcl_context_t context = rcl_get_zero_initialized_context();
+    const char * some_args[] = {"some-arg"};
+    ret = rcl_init(0, some_args, &init_options, &context);
+    EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret);
+    rcl_reset_error();
+    ASSERT_FALSE(rcl_context_is_valid(&context));
+  }
+  {
+    // If an invalid ROS arg is given, init should fail.
+    rcl_context_t context = rcl_get_zero_initialized_context();
+    const char * bad_remap_args[] = {
+      "some-arg", RCL_ROS_ARGS_FLAG, RCL_REMAP_FLAG, "name:="};
+    const size_t argc = sizeof(bad_remap_args) / sizeof(const char *);
+    ret = rcl_init(argc, bad_remap_args, &init_options, &context);
+    EXPECT_EQ(RCL_RET_INVALID_ROS_ARGS, ret);
+    rcl_reset_error();
+    ASSERT_FALSE(rcl_context_is_valid(&context));
+  }
+  {
+    // If an invalid enclave is given, init should fail.
+    rcl_context_t context = rcl_get_zero_initialized_context();
+    const char * bad_enclave_args[] = {
+      "some-arg", RCL_ROS_ARGS_FLAG, RCL_ENCLAVE_FLAG, "1foo"};
+    const size_t argc = sizeof(bad_enclave_args) / sizeof(const char *);
+    ret = rcl_init(argc, bad_enclave_args, &init_options, &context);
+    EXPECT_EQ(RCL_RET_ERROR, ret);
+    rcl_reset_error();
+    ASSERT_FALSE(rcl_context_is_valid(&context));
+  }
+  {
+    // If security keystore is invalid, init should fail.
+    ASSERT_TRUE(rcutils_set_env(ROS_SECURITY_ENABLE_VAR_NAME, "true"));
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+      EXPECT_TRUE(rcutils_set_env(ROS_SECURITY_ENABLE_VAR_NAME, ""));
+    });
+    ASSERT_TRUE(rcutils_set_env(ROS_SECURITY_STRATEGY_VAR_NAME, "Enforce"));
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+      EXPECT_TRUE(rcutils_set_env(ROS_SECURITY_STRATEGY_VAR_NAME, ""));
+    });
```

Comment:
Hmm, I'd think that if `ROS_SECURITY_ENCLAVE_OVERRIDE` is not set, we'll get `""` for it [here](https://github.com/ros2/rcl/blob/1cae3b6949bb47e1ca92ebb6c331a2b93b3f5205/rcl/src/rcl/security.c#L144), resulting in no override being used [here](https://github.com/ros2/rcl/blob/1cae3b6949bb47e1ca92ebb6c331a2b93b3f5205/rcl/src/rcl/security.c#L153) and [this](https://github.com/ros2/rcl/blob/1cae3b6949bb47e1ca92ebb6c331a2b93b3f5205/rcl/src/rcl/security.c#L182) being executed. In other words, it would use the keystore regardless, no?

Reply:
You are right, I misread the code initially